### PR TITLE
chore: add meta.yaml for ansible sample

### PR DESCRIPTION
**Pull request by svor**
### What does this PR do?
Registers **ansible-demo** sample. The sample was forked from https://github.com/kyetter/ansible-devspace-demo and it's located in https://github.com/devspaces-samples/ansible-devspaces-demo.

This sample has **Community** label and uses **quay.io/devspaces/ansible-creator-ee** as a tools image. The default branch that uses for the next DS version uses **latest** tag and branches with specific DS version use appropriate tag (e.g. devspaces-**3.8**-rhel-8 will use quay.io/devspaces/ansible-creator-ee:**3.8** image)
<details>
  <summary>The whole devfile.yaml is</summary>

 ```yaml
schemaVersion: 2.2.0
metadata:
  name: ansible-demo
components:
  - name: tooling-container
    container:
      image: quay.io/mloriedo/ansible-creator-ee:latest
      memoryRequest: 256M
      memoryLimit: 6Gi
      cpuRequest: 250m
      cpuLimit: 2000m
      args: ['tail', '-f', '/dev/null']
commands:
  - id: oc-install
    exec:
      commandLine: '/usr/local/bin/ansible-playbook ${PROJECT_SOURCE}/playbooks/install_oc.yml'
      workingDir: ${PROJECT_SOURCE}
      group:
        kind: build
        isDefault: true
      component: tooling-container
  - id: molecule-test
    exec:
      label: "Molecule: Run Scenario for Backup Role"
      commandLine: 'source $HOME/.bashrc && molecule test'
      workingDir: ${PROJECTS_ROOT}/ansible-devspace-demo/roles/backup_file
      group:
        kind: run
        isDefault: true
      component: tooling-container
  - id: molecule-verify
    exec:
      label: "Molecule: Validate Backup File Role"
      commandLine: 'source $HOME/.bashrc && molecule verify'
      workingDir: ${PROJECTS_ROOT}/ansible-devspace-demo/roles/backup_file
      group:
        kind: run
        isDefault: true
      component: tooling-container
events:
  postStart:
    - "oc-install"
```
</details>

<details>
  <summary>Dockerfile is located in the sample's repo and its content is</summary>

```dockerfile
FROM quay.io/ansible/creator-ee

ENV HOME /home/runner

# nodejs 16 + VSCODE_NODEJS_RUNTIME_DIR are required on ubi9 based images
# until we fix https://github.com/eclipse/che/issues/21778
# When fixed, we won't need this Dockerfile anymore.
# c.f. https://github.com/che-incubator/che-code/pull/120
RUN \
curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.3/install.sh | bash && \
export NVM_DIR="$HOME/.nvm" && \
[ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh" && \
nvm install 16.20.0
ENV VSCODE_NODEJS_RUNTIME_DIR="$HOME/.nvm/versions/node/v16.20.0/bin/"
```

</details>

### What issues does this PR fix or reference?
https://issues.redhat.com/browse/CRW-4359

### How to test?
1. Start a workspace using a [factory link for dev-sandbox](https://devspaces.apps.sandbox-m4.g2pi.p1.openshiftapps.com/dashboard/#/https://github.com/devspaces-samples/ansible-devspaces-demo)
2. When the workspace is ready, open **hello_world.yml** file. _Ansible-lint_ should validate the content and show an error.
3. Try to type something using _autocomplete_ feature 
4. Run commands from the devfile
-  **Molecule: Run Scenario for Backup Role** (will run whole scenario for backup role, the source code of this scenario is located in /projects/ansible-devspace-demo/roles/backup_file/molecule). As the result of this command you should see 4 new pods in the user namespace that were installed by ansible.
- **Molecule: Validate Backup File Role** (will run only **verify** molecule scenario on the instances)

### Screenshots

screenshot-devspaces apps ocp412-aws crw-qe com-2023 06 07-15_34_21

screenshot-devspaces apps ocp412-aws crw-qe com-2023 06 07-15_48_55

#### Release Notes

Add an Ansible sample into the devfile registry. The sample uses community image in the main component.

### CODE REVIEW

**Comment by nickboldt** [comment]
Not great to use `quay.io/mloriedo/ansible-creator-ee:latest` from https://github.com/devspaces-samples/ansible-devspace-demo/blob/devspaces-3-rhel-8/devfile.yaml#L7 in product samples. 

-1. We should at least copy that image to quay.io/devspaces/ org.

**Comment by nickboldt** [comment]
> +1 to have the image on quay.io/devspaces/ org. Since I don't have permissions, whom can I ask to push it to quay.io/devspaces? maybe @nickboldt or @mkuznyetsov ?

https://quay.io/repository/devspaces/ansible-creator-ee?tab=info created
Check email for invite, @svor

**Comment by svor** [comment]
@nickboldt thanks!
Now the sample is based on quay.io/devspaces/ansible-creator-ee:latest https://github.com/devspaces-samples/ansible-devspace-demo/blob/devspaces-3-rhel-8/devfile.yaml#LL7C14-L7C57

**Comment by nickboldt** [comment]
So we don't want a per-release-version tag for the sidecar / runtime image? Just `:latest` ? How do we guarantee it still works with 3.7 when we move to 3.8, and make changes to it?

**Comment by svor** [comment]
> So we don't want a per-release-version tag for the sidecar / runtime image? Just `:latest` ? How do we guarantee it still works with 3.7 when we move to 3.8, and make changes to it?

Ideally we should have a tag for each version but I'm not sure if we should do that if the sample is treated as a community

**Comment by l0rd** [comment]
> Ideally we should have a tag for each version but I'm not sure if we should do that if the sample is treated as a community

I would say that, community or not, using `latest` is a bad practice.

**Comment by svor** [comment]
> > Ideally we should have a tag for each version but I'm not sure if we should do that if the sample is treated as a community
> 
> I would say that, community or not, using `latest` is a bad practice.

Should we create a new tag for each DS release for https://quay.io/repository/devspaces/ansible-creator-ee or we can create a new tag when new changes are coming?

**Comment by l0rd** [comment]
> Should we create a new tag for each DS release for [quay.io/repository/devspaces/ansible-creator-ee](https://quay.io/repository/devspaces/ansible-creator-ee) or we can create a new tag when new changes are coming?

This really depends on what's simplest. Of course we don't need to re-tag if there are no changes but we need to evaluate if there is a risk that the check for changes can eventually fail and if it is consistent with the rest of the tagging.

**Comment by svor** [comment]
Added a GH action into the sample's repo https://github.com/devspaces-samples/ansible-devspace-demo/blob/devspaces-3-rhel-8/.github/workflows/image-build.yaml

With this action, each time when devstudio-release creates a new branch for a new version of DevSpaces (e.g. devspaces-3.8-rhel-8) in the sample repository - the image will be built and published with a new tag (e.g. 3.8) on https://quay.io/repository/devspaces/ansible-creator-ee. Also devfile.yaml in the repo will be updated to use appropriate image's tag.

The default branch devspaces-3-rhel-8 uses quay.io/devspaces/ansible-creator-ee:latest

File: product/tagRelease.sh [hunk 1]
```diff
@@ -276,6 +276,7 @@ php-hello-world \
 python-hello-world \
 quarkus-quickstarts \
 web-nodejs-sample \
+ansible-devspace-demo \
```
Comment (dmytro-ndp) [inline product/tagRelease.sh:279]: it's plural `devspaces`, not `devspace`

Comment (svor) [reply product/tagRelease.sh]: Good catch! Fixed

**Comment by svor** [comment]
@l0rd 

> @svor this is great, I was able to test it successfully. I have a couple of remarks:
> 
> * [The GitHub action uses `secrets.VSVYDENK_GITHUB_TOKEN`](https://github.com/devspaces-samples/ansible-devspace-demo/blob/devspaces-3.7-rhel-8/.github/workflows/image-build.yaml#L53). Is this intentional?

Yes, I added the token from my personal GitHub account to commit changes. We can use [che-bot](https://github.com/che-bot) or [devstudio-release](https://github.com/devstudio-release) bot for this operation. I just need to find someone who can add a secret with bot's token into the sample repo https://github.com/devspaces-samples/ansible-devspaces-demo

> * [In the Dockerfile the base image quay.io/ansible/creator-ee isn't referenced using a tag](https://github.com/devspaces-samples/ansible-devspace-demo/blob/devspaces-3.7-rhel-8/Dockerfile#L1). The workspace may get broken if the image is rebuild. I would rather use the sha (`quay.io/ansible/creator-ee@sha256:c89ecbcf47bfa956a2ed3c4939cd29a53298943c8db11eb35d20b9b939ba2a48`) or the tag (`quay.io/ansible/creator-ee:v0.18.0`)

Fixed. Now it uses the sha: https://github.com/devspaces-samples/ansible-devspaces-demo/blob/devspaces-3-rhel-8/Dockerfile#L2

**Review by l0rd — APPROVED** [review]

**Pull request merged**